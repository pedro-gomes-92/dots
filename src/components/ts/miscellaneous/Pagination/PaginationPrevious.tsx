import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { ButtonProps, Button } from '../../actions';

export interface PaginationPreviousProps extends ButtonProps {}

export const PaginationPrevious: FunctionComponent<PaginationPreviousProps> = ({
  className,
  ...rest
}: PaginationPreviousProps): JSX.Element => (
  <Button className={classnames('pagination-previous', className)} {...rest} />
);
PaginationPrevious.defaultProps = {
  icon: 'chevron-left-outline',
};
