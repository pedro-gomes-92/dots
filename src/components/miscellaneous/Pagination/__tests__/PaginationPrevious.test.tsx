import React from 'react';
import { render } from '@testing-library/react';

import { PaginationPrevious } from '../PaginationPrevious';

describe('Given <PaginationPrevious />', (): void => {
  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      const { container } = render(<PaginationPrevious onClick={(): void => {}} />);
      expect(container).toMatchSnapshot();
    });
  });
});
