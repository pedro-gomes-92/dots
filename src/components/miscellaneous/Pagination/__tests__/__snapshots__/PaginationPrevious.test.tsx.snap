// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <PaginationPrevious /> When initializes Then should match the snapshot 1`] = `
<div>
  <button
    class="button pagination-previous"
  >
    <span
      class="icon"
    >
      <i
        class="icon-font icon-font-chevron-left-outline"
      />
    </span>
  </button>
</div>
`;
